deployments/core.go

```diff
@@ -0,0 +1,255 @@
+		return nil
+	}
+
+	if out, err := helpers.KubectlDeleteEmbeddedYaml(coreDeploymentYamlPath, true); err != nil {
+		return errors.Wrap(err, fmt.Sprintf("Deleting %s failed:\n%s", coreDeploymentYamlPath, out))
+	}
+	if err = c.Kubectl.CoreV1().Secrets(coreDeploymentNamespace).Delete(context.Background(), coreSecretName, metav1.DeleteOptions{}); err != nil {
+		return errors.Wrapf(err, "Failed deleting secret %s", coreSecretName)
+	}
+
+	message := "Deleting Core component namespace " + coreDeploymentNamespace
+	_, err = helpers.WaitForCommandCompletion(ui, message,
+		func() (string, error) {
+			return "", c.DeleteNamespace(coreDeploymentNamespace)
+		},
+	)
+	if err != nil {
+		return errors.Wrapf(err, "Failed deleting namespace %s", coreDeploymentNamespace)
+	}
+
+	message = "Waiting for Core namespace to be gone"
+	warning, err := helpers.WaitForCommandCompletion(ui, message,
+		func() (string, error) {
+			var err error
+			for err == nil {
+				_, err = c.Kubectl.CoreV1().Namespaces().Get(
+					context.Background(),
+					coreDeploymentNamespace,
+					metav1.GetOptions{},
+				)
+			}
+			if serr, ok := err.(*apierrors.StatusError); ok {
+				if serr.ErrStatus.Reason == metav1.StatusReasonNotFound {
+					return "", nil
+				}
+			}
+
+			return "", err
+		},
+	)
+	if err != nil {
+		return err
+	}
+	if warning != "" {
+		ui.Exclamation().Msg(warning)
+	}
+
+	ui.Success().Msg("Core component removed")
+
+	return nil
+}
+
+// Create kubernetes namespace for core component
+func (core Core) createNamespace(c *kubernetes.Cluster, ui *ui.UI) error {
+	if _, err := c.Kubectl.CoreV1().Namespaces().Create(
+		context.Background(),
+		&corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: coreDeploymentNamespace,
+				Labels: map[string]string{
+					"quarks.cloudfoundry.org/monitored": "quarks-secret",
+					kubernetes.FusemlDeploymentLabelKey: kubernetes.FusemlDeploymentLabelValue,
+				},
+			},
+		},
+		metav1.CreateOptions{},
+	); err != nil {
+		return nil
+	}
+
+	if err := c.LabelNamespace(coreDeploymentNamespace, kubernetes.FusemlDeploymentLabelKey, kubernetes.FusemlDeploymentLabelValue); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Create secret to store gitea credentials
+// This might be different gitea instance than the one deployed by fuseml-installer!
+func (core Core) createCoreCredsSecret(c *kubernetes.Cluster, domain string) error {
+
+	// TODO if gitea information were not provided, read them from gitea deployment in this cluster?
+	giteaUsername, exists := os.LookupEnv("GITEA_USERNAME")
+	if !exists {
+		//return errors.New("Value for gitea user name (GITEA_USERNAME) was not provided.")
+		giteaUsername = "dev"
+	}
+	giteaPassword, exists := os.LookupEnv("GITEA_PASSWORD")
+	if !exists {
+		// return errors.New("Value for gitea user password (GITEA_PASSWORD) was not provided.")
+		giteaPassword = "changeme"
+	}
```

Comment:
Instead of hard-coding the default values yet again, couldn't you reference some common configuration both in here and in the gitea deployment ? If you want to keep the env vars override logic, it should also go in the same common configuration, otherwise the values won't be reflected in the gitea deployment.

Reply:
Yes, either common configuration place, or actually reading these values from existing gitea secret. Or both. I was planning it, that's why there's a TODO comment there...

Reply:
Reading the values from the gitea secret is a good alternative, thank you.